homeassistant/components/tasmota/light.py

```diff
@@ -0,0 +1,253 @@
+"""Support for MQTT JSON lights."""
+import logging
+
+from hatasmota.light import (
+    LIGHT_TYPE_COLDWARM,
+    LIGHT_TYPE_NONE,
+    LIGHT_TYPE_RGB,
+    LIGHT_TYPE_RGBCW,
+    LIGHT_TYPE_RGBW,
+)
+
+from homeassistant.components import light
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_EFFECT,
+    ATTR_HS_COLOR,
+    ATTR_TRANSITION,
+    ATTR_WHITE_VALUE,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_COLOR_TEMP,
+    SUPPORT_EFFECT,
+    SUPPORT_TRANSITION,
+    SUPPORT_WHITE_VALUE,
+    LightEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+import homeassistant.util.color as color_util
+
+from .const import DOMAIN as TASMOTA_DOMAIN
+from .discovery import TASMOTA_DISCOVERY_ENTITY_NEW
+from .mixins import TasmotaAvailability, TasmotaDiscoveryUpdate
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_BRIGHTNESS = False
+DEFAULT_COLOR_TEMP = False
+DEFAULT_EFFECT = False
+DEFAULT_FLASH_TIME_LONG = 10
+DEFAULT_FLASH_TIME_SHORT = 2
+DEFAULT_NAME = "MQTT JSON Light"
+DEFAULT_OPTIMISTIC = False
+DEFAULT_RGB = False
+DEFAULT_WHITE_VALUE = False
+DEFAULT_XY = False
+DEFAULT_HS = False
+
+CONF_EFFECT_LIST = "effect_list"
+
+CONF_FLASH_TIME_LONG = "flash_time_long"
+CONF_FLASH_TIME_SHORT = "flash_time_short"
+CONF_HS = "hs"
+
+DEFAULT_BRIGHTNESS_MAX = 255
+TASMOTA_BRIGHTNESS_MAX = 100
+MIN_MIREDS = 153
+MAX_MIREDS = 500
+
```

Comment:
SetOption82 Should be included on option list:
```
Reduce the CT range from 153..500 to 200.380 to accomodate with Alexa range
0 = CT ranges from 153 to 500 (default)
1 = CT ranges from 200 to 380 (although you can still set in from 153 to 500)
```
TuyaMCU normally have a reduced range too.

Reply:
Fixed!

Reply:
Fixed, thanks!